site/content/blog/2023-09-04-ds-store-parsing/index.md

```diff
@@ -0,0 +1,280 @@
+---
+title: "Parsing .DS_Store files with ZAP"
+summary: "ZAP Spider can now probe and parse macOS' .DS_Store files."
+images:
+type: post
+tags:
+- blog
+- recon
+- spider
+date: "2023-09-04"
+authors:
+    - skyper
+---
+macOS has been using [`.DS_Store`][wikipedia-ds-store] hidden files for many
+years to store metadata about files and folders on a file system. These metadata
+are mainly created and used by the [Finder][wikipedia-finder] to store
+attributes like your preferred view option for a particular folder, or the icon
+positions of the files it contains.
+
+`.DS_Store` files, when accidentally published and accessible on the Internet,
+can leak information about the structure of a file hierarchy, leading to
+potential information disclosure vulnerabilities. It is often the case when
+stored in the Git repository of a website.
+
+Discovering and reading the contents of `.DS_Store` files present on a website
+can prove to be a valuable asset when conducting web security audits and
+penetration tests. This has led me to the idea of integrating this
+reconnaissance task as part of the ZAP Spider tool, enabling the process to be
+fully automated and saving time during security engagements. But first things
+first, one needs to parse the internal structure of such files.
+
+## Parsing .DS_Store files
+
+The specifications detailing the internal structure of `.DS_Store` files have
+never been made public, but [thanks to community-driven
+efforts][0day-parsing-ds-store], most parts have been reverse engineered,
+enabling people to create custom parsers. Nevertheless, writing a parser for
+`.DS_Store` files is not a trivial task given the complexity of the format. This
+is why not much implementations can be found online, and most of them are in
+Python, Go and Perl only.
```

Comment:
```suggestion
is why few implementations can be found online, and most of them are in
Python, Go, or Perl only.
```

Reply:
Agreed. It reads better this way. Thanks.